src/eth_provider/utils.rs

```diff
@@ -88,4 +90,28 @@ mod tests {
             doc! {"test_key": format!("0x{}", "0".repeat(64))}
         );
     }
+
+    #[test]
+    fn test_split_u256() {
+        // Define a property to test the split_u256 function
+        proptest!(|(value in any::<B256>())| {
+            // Convert the B256 value to a hexadecimal string
+            let hex_string = format!("{:?}", value);
+            // Convert the hexadecimal string to a U256 value
+            let u256_value = U256::from_str(&hex_string).unwrap();
+
+            // Call the split_u256 function
+            let result = split_u256::<FieldElement>(u256_value);
+
+            // Format the combined hexadecimal string with different widths for each part
+            let combined_hex = format!("{:#0width1$x}{:0width2$x}",
+                u128::from_str(&result[1].to_string()).unwrap(),
+                u128::from_str(&result[0].to_string()).unwrap(),
+                width1 = 34,
+                width2 = 32);
```

Comment:
If you make the above change, you can then reconstruct a U256 (or B256 if you can't do `any::<U256>`) and compare the results instead of the string.